docs/bigtable-cluster-api.rst

```diff
@@ -0,0 +1,175 @@
+
+.. note::
+
+    When modifying a cluster (via a `CreateCluster`_, `UpdateCluster`_ or
+    `UndeleteCluster`_ request), the Bigtable API will return a long-running
+    `Operation`_. This will be stored on the object after each of
+    :meth:`create() <gcloud.bigtable.cluster.Cluster.create>`,
+    :meth:`update() <gcloud.bigtable.cluster.Cluster.update>` and
+    :meth:`undelete() <gcloud.bigtable.cluster.Cluster.undelete>` are called.
+
+You can check if a long-running operation (for a
+:meth:`create() <gcloud.bigtable.cluster.Cluster.create>`,
+:meth:`update() <gcloud.bigtable.cluster.Cluster.update>` or
+:meth:`undelete() <gcloud.bigtable.cluster.Cluster.undelete>`) has finished
+by making a `GetOperation`_ request with
+:meth:`Operation.finished() <gcloud.bigtable.cluster.Operation.finished>`:
+
+.. code:: python
+
+    >>> operation = cluster.create()
+    >>> operation.finished()
+    True
+
+.. note::
+
+    Once an :class:`Operation <gcloud.bigtable.cluster.Operation>` object
+    has returned :data:`True` from ``finished()``, the object should not
+    be re-used. Subsequent calls to
+    :meth:`finished() <gcloud.bigtable.cluster.Operation.finished>`
+    will result in a :class:`ValueError <exceptions.ValueError>`.
+
+Get metadata for an existing Cluster
+------------------------------------
+
+After creating the cluster object, make a `GetCluster`_ API request
+with :meth:`reload() <gcloud.bigtable.cluster.Cluster.reload>`:
+
+.. code:: python
+
+    cluster.reload()
+
+This will load ``serve_nodes`` and ``display_name`` for the existing
+``cluster`` in addition to the ``cluster_id``, ``zone`` and ``project``
+already set on the :class:`Cluster <gcloud.bigtable.cluster.Cluster>` object.
+
+Update an existing Cluster
+--------------------------
+
+After creating the cluster object, make an `UpdateCluster`_ API request
+with :meth:`update() <gcloud.bigtable.cluster.Cluster.update>`:
+
+.. code:: python
+
+    client.display_name = 'New display_name'
+    cluster.update()
+
+Delete an existing Cluster
+--------------------------
+
+Make a `DeleteCluster`_ API request with
+:meth:`delete() <gcloud.bigtable.cluster.Cluster.delete>`:
+
+.. code:: python
+
+    cluster.delete()
+
+Undelete a deleted Cluster
+--------------------------
+
+Make an `UndeleteCluster`_ API request with
+:meth:`undelete() <gcloud.bigtable.cluster.Cluster.undelete>`:
+
+.. code:: python
+
+    cluster.undelete()
+
+Next Step
+---------
+
+Now we go down the hierarchy from
+:class:`Cluster <gcloud.bigtable.cluster.Cluster>` to a
+:class:`Table <gcloud.bigtable.table.Table>`.
+
+Head next to learn about the :doc:`bigtable-table-api`.
+
+.. _Cluster Admin API: https://cloud.google.com/bigtable/docs/creating-cluster
+.. _CreateCluster: https://github.com/GoogleCloudPlatform/cloud-bigtable-client/blob/2aae624081f652427052fb652d3ae43d8ac5bf5a/bigtable-protos/src/main/proto/google/bigtable/admin/cluster/v1/bigtable_cluster_service.proto#L66-L68
```

Comment:
Yes. The line numbers won't be useful if I don't pin to a known point in history.